robotoff/workers/update_listener.py

```diff
@@ -102,16 +106,43 @@ def process_redis_update(self, redis_update: RedisUpdate):
                 )
 
 
+@backoff.on_exception(
+    backoff.expo,
+    ConnectionError,
+    max_tries=10,
+    max_time=300,  # 5 minutes
+    jitter=backoff.random_jitter,
+    on_backoff=lambda details: logger.error(
+        "Redis connection error (attempt %d/%d): %s. Retrying in %.1f seconds...",
+        details["tries"],
+        10,
+        details["exception"],
+        details["wait"],
+    ),
+    on_giveup=lambda details: logger.critical(
+        "Max retries (%d) reached. Update listener is terminating.", details["tries"]
+    ),
+)
 def run_update_listener():
     """Run the update import daemon.
 
     This daemon listens to the Redis stream containing information about
     product updates and triggers appropriate actions.
     """
-    redis_client = get_redis_client()
-    update_listener = UpdateListener(
-        redis_client=redis_client,
-        redis_stream_name=settings.REDIS_STREAM_NAME,
-        redis_latest_id_key=settings.REDIS_LATEST_ID_KEY,
-    )
-    update_listener.run()
+    while True:
+        try:
+            logger.info("Starting Redis update listener...")
+            redis_client = get_redis_client()
+            update_listener = UpdateListener(
+                redis_client=redis_client,
+                redis_stream_name=settings.REDIS_STREAM_NAME,
+                redis_latest_id_key=settings.REDIS_LATEST_ID_KEY,
+            )
+            update_listener.run()
+        except Exception as e:
+            logger.critical(
+                "Unexpected error in update listener: %s", str(e), exc_info=True
+            )
+            # for non-connection errors, wait a bit before retrying
+            time.sleep(5)
```

Comment:
No need to have this block, as we're catching here any error, including ConnectionError. It's good to use logger.critical to log it in Sentry though, so we can keep the `while True` loop.
```suggestion
```